stdlib/public/core/Integers.swift

```diff
@@ -1300,6 +1300,152 @@ extension BinaryInteger {
     return self.magnitude % other.magnitude == 0
   }
 
+  /// Returns `true` if this value is a power of the given base, and `false`
+  /// otherwise.
+  ///
+  /// For two integers *a* and *b*, *a* is a power of *b* if *a* is equal to
+  /// any repeated multiplication of *b*. For example, *16* is a power of *2*
+  /// because _6 = 2 * 2 * 2 * 2_. *-27* is a power of *-3* because
+  /// _-27 = -3 * -3 * -3_. *1* is a power of everything because any number to
+  /// the zero power is one.
+  ///
+  /// For the corner case where base is zero, `x.isPower(of: 0)` is `true` if
+  /// `x` is either zero or one, and `false` otherwise.
+  ///
+  /// - Parameter base: The base value to test.
+  @inlinable
+  public func isPower(of base: Self) -> Bool {
+    // Fast path when base is one of the common cases.
+    if base == 2 { return self._isPowerOfTwo }
+    if base == 10 { return self._isPowerOfTen }
+    if base._isPowerOfTwo { return self._isPowerOf(powerOfTwo: base) }
```

Comment:
We have a "fast path" for some commonly used bases, but this doesn't necessarily mean the other bases are so infrequent that we should make them slow.

From your link above:

> ... it can actually degrade performance in non-intuitive ways as it marks all other code (including subsequent _fastPaths) as being cold. Consider _fastPath as basically spraying the rest of the code with a Mr. Freeze-style ice gun.

Reply:
Thanks for catching those typos, Ben!

I agree with Michel that "slow path" in this context means _relatively inefficient algorithm_ only, but not *cold path*.

This patch is available to be included into other modules (e.g., swift-preview), since it does not access existing internal implementation of standard library.